File: restart-thread/android/shared/src/commonMain/kotlin/com/restartthread/shared/domain/DeterministicRecovery.kt
package com.restartthread.shared.domain

object DeterministicRecovery {
    const val SAFE_FIRST_ACTION =
        "Review your saved words, then choose one small, reversible next step."

    fun fromText(text: String): RecoveryDraft {
        require(text.isNotBlank())
        return RecoveryDraft(
            startHere = SAFE_FIRST_ACTION,
            evidence = text.trim(),
            isGenerated = false,
        )
    }

    fun fromSavedVoice(): RecoveryDraft = RecoveryDraft(
        startHere = "Listen to your saved note, then choose one small, reversible next step.",
        evidence = "Voice note saved on this device; no transcript was created.",
        isGenerated = false,
    )
}

File: restart-thread/apple/Sources/RestartThreadCore/Domain/DeterministicRecovery.swift
import Foundation

public enum DeterministicRecovery {
    public static let safeFirstAction =
        "Review your saved words, then choose one small, reversible next step."

    public static func fromText(_ text: String) -> RecoveryDraft? {
        let captured = text.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !captured.isEmpty else { return nil }
        return RecoveryDraft(
            startHere: safeFirstAction,
            evidence: captured,
            isGenerated: false
        )
    }

    public static func fromSavedVoice() -> RecoveryDraft {
        RecoveryDraft(
            startHere: "Listen to your saved note, then choose one small, reversible next step.",
            evidence: "Voice note saved on this device; no transcript was created.",
            isGenerated: false
        )
    }
}
